### DISCUSSION TITLE
no working examples of useWriteContract in docs

### DISCUSSION THREAD

**Discussion by clifton** [wevm/wagmi#76] [Q&A] [answered]
I'm having problems getting `useWriteContract` to work. are there any canonical examples? I'm also having to cast `{ value: <eth> } as Overrides` if i want to call a function passing eth. https://wagmi-xyz.vercel.app/docs/hooks/useContractWrite is an incomplete example.

Is there a particular reason why you went with the current API design over just `const contract = useContract<ContractType>(...); await contract.mint(...);` as it exists in react-web3?

**Comment by clifton** [comment]
it seems like the ideal way to do this is to use `useContract<..>(..)` with `userSigner()` and then just call the methods directly. i suppose `useContractRead|Write` is just for contracts which u do not typechain artifacts for?

**Comment by jasheal** [comment, marked as answer]
I posted an example here https://github.com/tmm/wagmi/discussions/49#discussioncomment-1977456
